modules/table/src/test/java/org/apache/ignite/internal/table/distributed/TableManagerRecoveryTest.java

```diff
@@ -332,9 +334,17 @@ private void startComponents() throws Exception {
 
         doReturn(nullCompletedFuture())
                 .when(replicaMgr).startReplica(any(RaftGroupEventsListener.class), any(), anyBoolean(), any(), any(), any(), any(), any());
-        doReturn(nullCompletedFuture())
-                .when(replicaMgr)
-                .startReplica(any(ReplicationGroupId.class), any(), any(), any(), any(), any(), anyBoolean(), any(), any());
+
+        ZonePartitionReplicaListener zonePartitionReplicaListener = mock(ZonePartitionReplicaListener.class);
+        Replica replica = mock(Replica.class);
+        doAnswer(invocation -> {
```

Comment:
Why do we use this backward mocking approach here? Is `replicaMgr` a spy object? I'm pretty sure that it's not and we can mock normally with `when().then()`